ucsmsdk/ucscoreutils.py

```diff
@@ -483,3 +483,174 @@ def get_naming_props(rn_str, rn_pattern):
+        children = sorted(MO_CLASS_META[meta_class_id].children)
+        total = len(children)
+        count = 0
+        if break_level is None or level < break_level + 1:
+            for child in children:
+                count += 1
+
+                if last_child:
+                    ancestor_str_ = ancestor_str + "   "
+                else:
+                    ancestor_str_ = ancestor_str + "  |"
+
+                _show_tree(child, break_level, level, ancestor_str_, ancestor,
+                           total == count)
+
+        ancestor.pop(index-1)
+
+
+def search_class_id(class_id):
+    """
+    case insensitive search for class_id in meta.
+    if unable to find exact class_id, this will also suggest matching class_id.
+
+    Args:
+        class_id (str): string matching class_id.(case insensitive)
+
+    Returns:
+        (str) or None
+
+    Example:
+        class_ids = search_class_id(class_id="ls")
+    """
+
+    from . import ucsmeta
+
+    meta_class_id = find_class_id_in_mo_meta_ignore_case(class_id=class_id)
+
+    if meta_class_id is not None:
+        return meta_class_id
+
+    # if class_id not exists in meta
+    l_class_id = class_id.lower()
+    class_ids = sorted([cid for cid in ucsmeta.MO_CLASS_ID
+                        if re.search(l_class_id, cid, re.IGNORECASE)])
+    if class_ids:
+        log.info("<%s> is not present.\n"
+                 "Related available class_ids are:\n%s" %
+                 (class_id, "\n".join(class_ids)))
+    else:
+        log.info("<%s> is not present." % class_id)
+
+
+def get_meta_info(class_id, include_prop=True,
+                  show_tree=True, break_level=None):
+    """
+    Gets class id meta information
+
+    Args:
+        class_id (str): string matching class_id.(case insensitive)
+        include_prop (bool): by default True. If False, excludes property.
+        show_tree (bool): by default True. If False will not display mo tree.
+        break_level (int): display the hierarchy till respective level.
+
+    Returns:
+        ClassIdMeta Object: class_id
+                            xml_attribute
+                            rn
+                            min_version
+                            access
+                            access_privilege
+                            parents : parent list
+                            children : children list
+                            properties : property list
+                            props : {property_name : MoPropertyMeta Object}
+
+    Example:
+        meta = get_meta_info(class_id="lsserver")
+        meta = get_meta_info(class_id="lsserver", break_level=2)
+        meta = get_meta_info(class_id="lsserver", include_prop=False)
+        meta = get_meta_info(class_id="lsserver", show_tree=False)
+
+        print(meta.xml_attribute)
+        print(meta.children)
+        print(meta.props["name"])
+    """
+
+    meta_class_id = search_class_id(class_id)
+    if not meta_class_id:
```

Comment:
@ragupta-git no need of an exception here, a return is enough. search_class_id throws the necessary msg.